### PR TITLE
Add introduction before "Table of contents"

### PR DESCRIPTION
**Pull request by ssurbhi560**
towards #1673 

PR Checklist:

- [x] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)

### CODE REVIEW

**Comment by ssurbhi560** [comment]
@kathatherine @beckermr @BastianZim, I am still working on improving the "what is conda-forge?" section before opening the PR, but I would like to get some feedback from you folks on this.

File: src/index.rst [hunk 1]
```diff
@@ -3,6 +3,31 @@
...
+
+Conda-forge is a community effort that provides conda packages for a wide range of software.
+
+**Missing a package that you would love to install with conda?**
+
+Chances are we have already packaged it for you. You can `search <https://anaconda.org/>`__ for packages online. Look out for packages provided by our ``conda-forge`` organization.
+
+**Cannot find a package or only outdated versions of a package?** - Everybody is welcome to contribute to our package stack!
+
+- To get started contributing packages see here :ref:`becoming_involved`.
```
Comment (kathatherine) [inline src/index.rst:20]: Need a comma after "packages".

Comment (kathatherine) [reply src/index.rst:20]: Also, the ref link needs to be ```see :ref:`here <becoming_involved>`.```

File: src/index.rst [hunk 2]
```diff
@@ -3,6 +3,31 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
```
Comment (kathatherine) [inline src/index.rst:4]: If we're adding text that isn't the "toctree" code, we should probably delete this text above, right? Did you build it locally and see if this text appears?

Comment (ssurbhi560) [reply src/index.rst:4]: That's right, @kathatherine, thanks for pointing it out, I will remove it. And yes I checked it locally. This is how it looks as of now: 
Screenshot 2022-04-12 at 3 59 00 AM

Comment (kathatherine) [reply src/index.rst:4]: Awesome, thanks!

File: src/index.rst (hunk 1 above)
Comment (ssurbhi560) [reply src/index.rst]: @kathatherine, actually the link works fine without the angle brackets, and if I do add them it giving an error. I think it is how pages, which are already in the documentation are linked and brackets are used for the external links.

**Review by kathatherine — APPROVED** [review]
Perfect, thanks!

**Review by beckermr — APPROVED** [review]

**Pull request merged**